### PR TITLE
Display author name in NotePreview 

### LINKED ISSUE #21: Track prestige
Keep track of the aggregation of "rating" scores for all of the notes a user has submitted, and display this value on their profile page.
- [x] Compute prestige score based "rating" score of all user-submitted notes.
- [x] Store prestige score in database with the user profile
- [x] Display the prestige score on hover of the author name

### PR DESCRIPTION
**Pull request by keelywan**
- Moved `constructUserFromSqlResult` from `UserRegistrationServlet` to the User Service class and added `getUserById` function
- Added two getter methods in the User model 
  - `getDisplayName`
  - `getPoints`
## GetAuthorInfoServlet 
- Calls `getUserById` function to retrieve all the user's information
- Returns only the necessary data (display name and points) so the front end can't see private data like user_id

## NotePreview 
- Display's the author's name
preview1

- When the display name is hovered over, a tooltip will pop up showing the number of points that user has
preview2

### CODE REVIEW

**Review by Aradhya-B — APPROVED** [review]
LGTM :)

**Review by andres-holguin — APPROVED** [review]
Looks good!

File: src/main/java/com/google/starfish/servlets/GetAuthorInfoServlet.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        res.getWriter().println(json);
+      }
+    } catch (SQLException ex) {
+      res.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+      res.getWriter().println("INTERNAL SERVER ERROR: " + ex);
+    }
+  }
+
+  /** Converts an object to JSON */
+  private String convertObjectToJSON(HashMap<String, String> user) {
```
Comment (cftorres) [inline src/main/java/com/google/starfish/servlets/GetAuthorInfoServlet.java:45]: I see this kind of logic in a lot of your servlets-- maybe you could use from Util file (I think Aradhya made this in a recent PR)?

**Review by cftorres — APPROVED** [review]
Nice-- very clean!

File: src/main/java/com/google/starfish/servlets/GetAuthorInfoServlet.java (hunk 1 above)
Comment (Aradhya-B) [reply src/main/java/com/google/starfish/servlets/GetAuthorInfoServlet.java:45]: Yep, I think we can make a new PR that includes a lot of these duplicated methods inside the `Utils` class once #72 gets merged!

**Review by nickKelman — APPROVED** [review]

**Pull request merged**